app/src/main/java/de/westnordost/streetcomplete/data/osmnotes/AvatarsDownloader.kt

```diff
@@ -42,11 +47,14 @@ class AvatarsDownloader(
     }
 
     /** download avatar for the given user and a known avatar url */
-    fun download(userId: Long, avatarUrl: String) {
+    suspend fun download(userId: Long, avatarUrl: String) {
         if (!ensureCacheDirExists()) return
         try {
             val avatarFile = File(cacheDir, "$userId")
-            URL(avatarUrl).saveToFile(avatarFile)
+            val response = httpClient.get(avatarUrl) {
+                expectSuccess = true
+            }
+            response.bodyAsChannel().copyAndClose(avatarFile.writeChannel())
```

Comment:
I've seem to remember that I read somewhere, that Ktor Client is based on / uses Okio because it is multiplatform.

Is it (not) possible to replace the dependency to `java.io.File` here and elsewhere? (E.g. the photo uploader also accesses the file system IIRC)

Reply:
I was also thinking that removing `java.io.File` would be a useful thing to do, but I'm also concious that this PR is getting big, and replacing `File` usage could itself be an independent bit of work.

I'll keep the Java API for now, and will do some investigations into Okio separately.

Reply:
Right, ok